### ISSUE TITLE
DE-DCA/UN-DCA enhancement (with profit if possible)

### ISSUE THREAD

**Issue by Masterofcoinak** [taniman/profit-trailer-enhancements#320]
Hallo devs,
I am pretty sure you got already ideas/requests for an UN-DCA upgrade/enhancement/add-on.... 
We can also call it DE-DCA. 

In times like now, everybody is carrying around huge bags and a lot of us don't have the budget to DCA out of this problem.  A simple way would be to sell back DCA steps. 
We could always do that in the exchange, but hopefully PT will do it for us in a smart way.
I thought about 3 ways:

1. Simple version: 
Just sell the amount coins of the previous DCA according the current market price. A kind of panic_sell_function for the last DCA step.
+ Balance is free
 - Bag is back in the even deeper none profit zone (in %)

2. Improved version: 
Establish a separate De-DCA sector in PT with separate setting file.
(PBL > PAIRS > DCA > UNDCA). Here we could set a selling strategy for the last DCA step.
The last DCA step would be treated as separate sale. (also goes like this into the sales log)
Base should be always buy amount and price of the last DCA.
+ Balance is free
+ Profit for an additional trade 
- Bag is back in the even deeper none profit zone (in %)

3. Advanced version: 
Same as above version, but the profit of the DCA sell will be used to improve the average price of your bag. The sell will be not shown in the sales log (or could be, but with 0% Profit). In case you do it more often, then this should make it possible to resolve your bag easier.
+ Balance is free
+ Bag is not as deep as before in the none profit zone (in %)

3.1 If you fix the free balance to its coin/pair, than it could just DCA again.
+ Improve the value of your pair by swing trade (DCA>UNDCA>DCA>UNDCA…)
And rebuy DCA at an improved low point

3.2 Or you just let the bot decide where to use the free balance. Maybe even make a rebuy timeout for the previous pair.
+ Bot could start to trade again and invests where it is best suited.

Getting huge and expensive bags can of course be stopped of your bot by limited DCA steps or Max DCA cost , but I think I am not the only who is releasing an additional DCA step in the hope of getting rid of the bag.

Sending it to the pending log by trigger is probably also a solution of some users, but this just lets trade the coin again. Bag remains as it is and frees no balance.

I know it doesn't make sense, but below are future settings. hahah ;).

#First we define the parameter to be in the UNDCA.
DEFAULT_UNDCA_enabled = false/true/-20
DEFAULT_UNDCA_min_cost = 1.0
#Than simple sell strategy as always
DEFAULT_UNDCA_A_sell_strategy = GAIN
DEFAULT_UNDCA_A_sell_value = 2.0
DEFAULT_UNDCA_B_sell_strategy = RSI
DEFAULT_UNDCA_B_sell_value = 55
DEFAULT_UNDCA_trailing_profit = 0.5
DEFAULT_UNDCA_max_profit = 0
DEFAULT_UNDCA_min_buy_volume = 100
DEFAULT_UNDCA_min_order_book_volume_percentage = 125
#normally it will go back to DCA
DEFAULT_DCA_rebuy_timeout = 60

Please think about it.

**Comment by Masterofcoinak** [comment]
I thought more and more about the my UN-DCA solution and meanwhile I think, it could be even used as a complete strategy. I hope i am not completely wrong. See above.

**Comment by DemoFreak** [comment]
@clownfish44 I guess you should re-read the OP.
It's absolutely beyond question that one should calculate the max DCA he could afford.

But as it's also undisputable that noone can predict the future, it would be a nice-to-have if PT could manage a partially sell of bags without need to login to the exchange. This could be done with a simple "DeDCA" button next to each pair.

And as a sugar on top then this automatic DeDCA triggers could be implemented to make it even better.

tl;dr:
- "DeDCA" button next to each pair in DCA log
- "DePending 50%" button next to each pair in Pending log
- automatic triggers that manage this de-DCA and de-Pending of pairs
